app/src/main/java/mozilla/lockbox/presenter/AutofillRoutePresenter.kt

```diff
@@ -146,6 +156,14 @@ class AutofillRoutePresenter(
         }
     }
 
+    private fun <T : DialogFragment> dismissDialogIfPresent(clazz: Class<T>) {
+        val fragmentManager = activity.supportFragmentManager
+        val presentedDialog = fragmentManager.findFragmentByTag(clazz.name)
+        log.info("presented dialog: $presentedDialog")
```

Comment:
In general, I don't like that the route needs to have fore-knowledge of the dialog that we're moving from.

Additionally, I'm not entirely sure if this is only working accidentally. I think I'd prefer to see a 

```kotlin
<T : Fragment> fun FragmentManager.findChildFragmentByClass(clazz: Class<T>)
```

which looks at `fragment.childFragmentManager` of each fragment (currently, `showDialogFragment` uses the `NavHostFragment` as the `currentFragment`).

Reply:
I don't know if I'd call this "accidental" but I would call it "hacky" ;)

it relies on us setting the tag for the dialog fragments to their respective class names as obtained by `clazz.name`.... I'm open to other solutions tho.